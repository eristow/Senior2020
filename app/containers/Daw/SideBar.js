import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { useInjectReducer } from 'utils/injectReducer';
import { makeSelectSideBarOpen } from './selectors';
import { showSideBar } from './actions';
import reducer from './reducer';

const OpenBtn = styled.button`
  margin-bottom: 3px;
  background: lightblue;
  border-radius: 5px;
`;

const Container = styled.div`
  position: absolute;
  height: 500px;
  width: 150px;
  border: 2px solid black;
  background: lightgray;
`;

const MenuBtn = styled.button`
  margin: 10px;
  width: 85%;
  background: lightblue;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2), 0 2px 4px 0 rgba(0, 0, 0, 0.19);
  border-radius: 5px;
`;

const sounds = ['Kick', 'Snare', 'Clap', 'Hi Hat Closed', 'Hi Hat Open'];

const key = 'daw';

export function SideBar({ openSideBar, sideBarOpen }) {
  useInjectReducer({ key, reducer });

  return (
    <>
      <OpenBtn type="button" onClick={openSideBar}>
        {!sideBarOpen ? '>>' : '<<'}
      </OpenBtn>
      {sideBarOpen ? (
        <Container>
          {sounds.map(sound => (
            <MenuBtn>{sound}</MenuBtn>
          ))}
        </Container>
      ) : (
        <div />
      )}
    </>
  );
}

SideBar.propTypes = {
  openSideBar: PropTypes.func,
  sideBarOpen: PropTypes.bool,
};

const mapStateToProps = createStructuredSelector({
  sideBarOpen: makeSelectSideBarOpen(),
});

const mapDispatchToProps = dispatch => ({
  openSideBar: () => {
    dispatch(showSideBar());
  },
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(SideBar);
