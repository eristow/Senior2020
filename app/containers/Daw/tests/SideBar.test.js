import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import { SideBar } from '../SideBar';

describe('<SideBar />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <SideBar dispatch={dispatch} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <SideBar />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
