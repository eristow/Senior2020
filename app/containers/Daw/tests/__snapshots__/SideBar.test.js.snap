// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> Should render and match the snapshot 1`] = `
<button
  class="SideBar__OpenBtn-sc-1ix0uck-0 fZxkvF"
  type="button"
>
  &gt;&gt;
</button>
`;
